doc/playerctl.1.in

```diff
@@ -1,123 +1,265 @@
+Multiple players can be specified in a comma-separated list.
+.It Fl l , -list-all
+List the names of running players that can be controlled.
+.It Fl p Ar NAME , Fl -player Ar NAME
+Control the specific player
+.Ar NAME .
+Multiple players can be specified in a comma-separated list.
+Defaults to the first available player.
+.It Fl V , -version
+Print version number, then exit.
+.El
+.Pp
+The commands are as follows:
+.Bl -tag -width Ds
+.It Cm status
+Get the current status of the player.
+.It Cm play
+Command the player to play.
+.It Cm pause
+Command the player to pause.
+.It Cm play-pause
+Command the player to toggle between play and pause.
+.It Cm stop
+Command the player to stop.
+.It Cm next
+Command the player to skip to the next track.
+.It Cm previous
+Command the player to skip to the previous track.
+.It Cm position Op Ar OFFSET Ns Op Ar + Ns | Ns Ar -
+Print the position of the current track in seconds.
+With
+.Ar OFFSET
+specified, seek to
+.Ar OFFSET
+seconds from the start of the current track.
+With the optional
+.Op Ar + Ns | Ns Ar -
+appended,
+seek forward or backward
+.Ar OFFSET
+seconds from the current position.
+.It Cm volume Op Ar LEVEL Ns Op + Ns | Ns Ar -
+Print the player's volume scaled from 0.0
+.Pq 0%
+to 1.0
+.Pq 100% .
+With
+.Ar LEVEL
+specified,
+set the player's volume to
+.Ar LEVEL .
+With the optional
+.Op Ar + Ns | Ns Ar -
+appended,
+increase or decrease the player's volume by
+.Ar LEVEL .
+.It Cm metadata Op Ar KEY
+Print available metadata information for the current track.
+When
+.Ar KEY is
+specified,
+print the value of
+.Ar KEY .
+.It Cm open Ar URI
+Open
+.Ar URI
+in the player.
+.Ar URI
+may be the name of a file or an external URL.
+.It Cm shuffle Op Ic On | Off
+Print the shuffle status of the player.
+With the shuffle status specified,
+set shuffle to either
+.Ic On
+or
+.Ic Off .
+.It Cm loop Op Ic None | Track | Playlist
+Print the loop status of the player.
+With the loop status specified,
+set the loop status to
+.Ic None
+.Pq disable looping ,
+.Ic Track
+.Pq loop the current track ,
+or
+.Ic Playlist
+.Pq loop the current playlist .
+.El
+.Ss Format Strings
+The output of the
+.Cm position ,
+.Cm metadata ,
+.Cm status
+and
+.Cm volume
+commands can be controlled using a format string.
+Variables set by these commands can be included in the format string
+by enclosing them in curly braces,
+i.e:
+.Brq Brq Va var .
+These will then be expanded on output.
+.Pp
+Each command has access to the following variables:
+.Bl -tag -width Ds
+.It Va playerName
+The name of the current player.
+.It Va position
+The time position of the current track,
+in microseconds.
+.It Va status
+The status of the current player.
+.It Va volume
+The player's volume scaled from 0.0
+.Pq 0%
+to 1.0
+.Pq 100% .
+.El
+.Pp
+Variables set by
+.Cm metadata
+are as follows:
+.Bl -tag -width Ds
+.It Va album
+The album of the current track.
+.It Va artist
+The artist of the current track.
+.It Va xesam:contentCreated
```

Comment:
All of the properties are optional and set by the player. `album` and `artist` are shortcuts for the `xesam:[prop]` version. They might not be present. Some of the properties like `mpris:trackid` and `mpris:length` I think are specified as required by mpris, but some players don't set them.

So with that in mind, I don't want to document anything that isn't portable across all players, like this one. I guess artist/album/title is ok because they're so common.

`position` and `playerName` and `volume` are set by playerctl from required properties not in the metadata so I'm ok with documenting those.

 Anything in the keys of `playerctl metadata` will be expanded.

See: https://freedesktop.org/wiki/Specifications/mpris-spec/metadata/

Reply:
I've truncated the list to just artist, album and title. How is in now?

Reply:
I'll add a link to the mpris-spec in SEE ALSO and reference it here.

Reply:
About the mpris-spec metadata page, is it an official list of all properties supported by the spec or is there more?

Reply:
There is supposedly an xesam spec out there somewhere but I can't find it and I've never seen it used outside of mpris. The spec links here, but the link is broken: http://www.xesam.org/main/XesamOntology.

Other players set their own metadata fields. Vlc sets `vlc:time`, `vlc:length`, and `vlc:publisher`.

Reply:
Looks like the internet archive has it last saved on [December 12, 2008](https://web.archive.org/web/20081212182150/http://www.xesam.org/main/XesamOntology). Not really relevant to the manpage itself, but I might follow this up with the freedesktop mailing lists.